### PR TITLE
Enable oss master to private main branch sync on GitHub Actions

### PR DESCRIPTION
**Pull request by yifanyang**

### CODE REVIEW

File: .github/workflows/private-mirror-sync.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Private Mirror Sync
```
Comment (vkryachko) [inline .github/workflows/private-mirror-sync.yml:1]: let's make sure it only runs on firebase/firebase-android-sdk repo.

Comment (yifanyang) [reply .github/workflows/private-mirror-sync.yml:1]: Done.

**Review by vkryachko — APPROVED** [review]

File: .github/workflows/private-mirror-sync.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           submodules: true
+          token: ${{ secrets.GOOGLE_OSS_BOT_TOKEN }}
       - name: Force push HEAD to private repo main branch
         run: |
           git remote add mirror git@github.com:FirebasePrivate/firebase-android-sdk.git
```
Comment (vkryachko) [inline .github/workflows/private-mirror-sync.yml:23]: how will `git push` know to use the right credentials?

Comment (yifanyang) [reply .github/workflows/private-mirror-sync.yml:23]: I just realized that I haven't actually tested it before sending it out for review. But it should work now IIUC.

> how will `git push` know to use the right credentials?

The `actions/checkout` [usage](https://github.com/actions/checkout#usage) section describes one flag `persist-credentials` which defaults to `true`. I think this flag makes following git invocations to use the above configured token.

**Pull request merged**